Tests/test_file_webp.py

```diff
@@ -122,6 +122,30 @@ def test_save_all(self, tmp_path):
             reloaded.seek(1)
             assert_image_similar(im2, reloaded, 1)
 
+    @skip_unless_feature("webp_anim")
+    def test_save_all_progress(self):
+        out = BytesIO()
+        progress = []
+
+        def callback(filename, frame_number, n_frames):
+            progress.append((filename, frame_number, n_frames))
+
+        Image.new("RGB", (1, 1)).save(out, "WEBP", save_all=True, progress=callback)
+        assert progress == [(None, 1, 1)]
+
+        out = BytesIO()
+        progress = []
+
+        with Image.open("Tests/images/iss634.webp") as im:
+            im2 = Image.new("RGB", im.size)
+            im.save(out, "WEBP", save_all=True, append_images=[im2], progress=callback)
+
+        expected = []
+        for i in range(42):
+            expected.append(("Tests/images/iss634.webp", i + 1, 43))
+        expected.append((None, 43, 43))
```

Comment:
Likewise, saying that the last frame comes from image #​2 would be more informative than "don't know" – especially when there's multiple images involved

Reply:
I've pushed a commit to switch to an index.